core/server/models/session.js

```diff
@@ -0,0 +1,49 @@
+const ghostBookshelf = require('./base');
+
+const Session = ghostBookshelf.Model.extend({
+    tableName: 'sessions',
+
+    parse: function parse(attrs) {
+        attrs.session_data = JSON.parse(attrs.session_data);
+        return attrs;
+    },
+
+    format: function format(attrs) {
+        if (attrs.session_data) {
+            attrs.session_data = JSON.stringify(attrs.session_data);
+        }
+        return attrs;
+    },
+
+    user: function () {
+        return this.belongsTo('User');
+    }
+}, {
+    setSession: function (id, data) {
```

Comment:
`models.Session.edit` would be consistent with other model update functions.
e.g. `models.Post.edit`

The `edit` endpoint in the base model already fetches the data.

Reply:
> return this.add({

Hm you either add or edit. Why is that?

Reply:
Ok 👍

Reply:
the only thing which is weird is

`models.Session.setSession`

We are setting a session on a session model 😆

Reply:
:shrimp: hows about `setSessionData`?

Reply:
But why putting the name "session" inside? Any reason? 🤔 I liked `upsert`.

Reply:
In general:

We always try to design model fn's consistently

e.g. `add: function add(data, unfilteredOptions) {`

The consistency helps with
- you can extend the fn like you want, you will always receive two objects
- consistency/readability

We try to avoid passing single attrs, otherwise you end up in

`fn(id, data)`
`fn(id, data, updated_at)`

`fn(1, null, Date)`

Reply:
Maybe look at some other model function examples. You should also pre-respect `options`.

Reply:
The reason is because the "upsert" was totally specific to sessions, so it seemed strange to give it a generic name. i'll update function sig and rename :+1: